cli/tsc_config.rs

```diff
@@ -48,6 +48,7 @@ const IGNORED_COMPILER_OPTIONS: [&str; 61] = [
   "inlineSourceMap",
   "inlineSources",
   "init",
+  "isolatedModules",
```

Comment:
Does this mean it can't be turned off? 😕

Reply:
Yes

Reply:
Although I guess this is not strictly required - I guess we can let you disable it. I wouldn't see why though.

Reply:
> Although I guess this is not strictly required - I guess we can let you disable it. I wouldn't see why though.

So people can suppress the breakages that would occur, for the short term at least.

For the long term... I'm not sure what our rule is for this. Do we allow any setting that isn't nonsensical in Deno? Or do we only allow settings justified with a use case? I think it was the former.

Reply:
> A static slice of all the compiler options that should be ignored that either have no effect on the compilation or would cause the emit to not work in Deno.

I guess disabling `isolatedModules` could cause the emit not to work if we stop using tsc for emit, and just do stripping via swc. The nice thing about defaulting here is that we can guarantee with some certainty that the swc emit will look like a tsc emit. This would also be useful for bundling.

I guess for this first pass we can remove this from the list so people can suppress breakages, and then once we actually have some code that makes the assumption that all typescript code works with `isolatedModules`, we can re-add it.

Reply:
The rule is "anything that has no impact or would have an undesired impact on the emit."  It doesn't have to break the emit.

That being said, introducing it in unstable is to get people to adapt to it, and there is certainly a use case where you want an unstable API but you can resolve all the `isolatedModules` fixes.  We should add it as an ignored option at some point post 1.5.0, like 1.6.0, IMO, but allow it to be changed up to that point.